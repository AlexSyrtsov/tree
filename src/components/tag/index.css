.tree-tag {
  padding: 5px 15px;
  background: darkgray;
  border-radius: 15px;
}
