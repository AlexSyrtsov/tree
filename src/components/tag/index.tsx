import { memo } from "react";

import { ITreeItemData } from "../../interfaces/interfaces";

import "./index.css";

interface ITagProps {
  tag: ITreeItemData;
}

export const Tag = memo(({ tag }: ITagProps) => {
  return (
    <div className="tree-tag">
      <span>{tag.tagTitle ?? tag.title}</span>
    </div>
  );
});
