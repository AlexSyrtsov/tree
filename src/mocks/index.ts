import { ITreeItemData } from "../interfaces/interfaces";

export const MOCK_TREE_ITEM_FIRST: ITreeItemData = {
  id: "0030d9c5-9f1b-431a-9080-03cc48dc117f",
  title: "Phones",
};

export const MOCK_TREE_ITEM_SECOND: ITreeItemData = {
  id: "0030d9c5-9f1b-431a-9080-03cc48dc117d",
  title: "Laptops",
};

export const MOCK_TREE_ITEM_THIRD: ITreeItemData = {
  id: "0030d9c5-9f1b-431a-9080-03cc48dc117z",
  title: "Desktop",
  parentId: "0030d9c5-9f1b-431a-9080-03cc48dc117f",
};

export const MOCK_TREE_ITEM_FOURTH: ITreeItemData = {
  id: "0030d9c5-9f1b-431a-9080-03cc48dc117a",
  title: "Watches",
  parentId: "0030d9c5-9f1b-431a-9080-03cc48dc117f",
};

export const MOCK_TREE_ITEM_DISABLED: ITreeItemData = {
  ...MOCK_TREE_ITEM_FIRST,
  disabled: true,
};

export const MOCK_TREE_ITEM_WITH_CHILDREN: ITreeItemData = {
  ...MOCK_TREE_ITEM_FIRST,
  childrenIds: [
    "0030d9c5-9f1b-431a-9080-03cc48dc117a",
    "0030d9c5-9f1b-431a-9080-03cc48dc117z",
  ],
};

export const MOCK_ITEMS_ARRAY = [
  MOCK_TREE_ITEM_WITH_CHILDREN,
  MOCK_TREE_ITEM_SECOND,
  MOCK_TREE_ITEM_THIRD,
  MOCK_TREE_ITEM_FOURTH,
];

export const MOCK_ITEMS_MAP = {
  "0030d9c5-9f1b-431a-9080-03cc48dc117f": MOCK_TREE_ITEM_WITH_CHILDREN,
  "0030d9c5-9f1b-431a-9080-03cc48dc117d": MOCK_TREE_ITEM_SECOND,
};
