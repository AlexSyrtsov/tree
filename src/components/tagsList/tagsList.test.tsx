import * as enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";

import { TagsList } from ".";
import { Tag } from "../tag";

import { MOCK_ITEMS_ARRAY } from "../../mocks";

enzyme.configure({ adapter: new Adapter() });

describe("TagsList component", () => {
  it("Match snapshot", () => {
    expect(<TagsList tags={MOCK_ITEMS_ARRAY} />).toMatchSnapshot();
  });
  it("Render corretly", () => {
    const result = enzyme.shallow(<TagsList tags={MOCK_ITEMS_ARRAY} />);
    expect(
      result.contains(<h4 className="tree__title">Selected variants</h4>)
    ).toBeTruthy();
    expect(result.find(Tag)).toHaveLength(4);
  });
});
