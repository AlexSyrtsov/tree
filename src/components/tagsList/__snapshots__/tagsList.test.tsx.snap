// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList component Match snapshot 1`] = `
<Memo
  tags={
    Array [
      Object {
        "childrenIds": Array [
          "0030d9c5-9f1b-431a-9080-03cc48dc117a",
          "0030d9c5-9f1b-431a-9080-03cc48dc117z",
        ],
        "id": "0030d9c5-9f1b-431a-9080-03cc48dc117f",
        "title": "Phones",
      },
      Object {
        "id": "0030d9c5-9f1b-431a-9080-03cc48dc117d",
        "title": "Laptops",
      },
      Object {
        "id": "0030d9c5-9f1b-431a-9080-03cc48dc117z",
        "parentId": "0030d9c5-9f1b-431a-9080-03cc48dc117f",
        "title": "Desktop",
      },
      Object {
        "id": "0030d9c5-9f1b-431a-9080-03cc48dc117a",
        "parentId": "0030d9c5-9f1b-431a-9080-03cc48dc117f",
        "title": "Watches",
      },
    ]
  }
/>
`;
